src/Microsoft.FeatureManagement.Telemetry.ApplicationInsights/TelemetryPublisherApplicationInsights.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+using Microsoft.ApplicationInsights;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.FeatureManagement.Telemetry.ApplicationInsights
+{
+    /// <summary>
+    /// Used to publish data from evaluation events to Application Insights
+    /// </summary>
+    public class TelemetryPublisherApplicationInsights : ITelemetryPublisher
```

Comment:
An application publishes telemetry to AppInsights. Is it right to call AppInsights a publisher?

Reply:
I don't think so. This is supposed to represent "This class is a publisher that pushes data to Application Insights". Jimmy had a suggestion to rearrange the name which might help (ie: `ApplicationInsightsTelemetryPublisher`)

Reply:
Right, AppInsights is a telemetry store, not a publisher. The name Jimmy suggested is better than the current one. I guess people prefer that one over `TelemetryPublisherForApplicationInsights`?

Reply:
I prefer `TelemetryPublisherForApplicationInsights`, as it's more clear. But I know people like to avoid conjunctions in class names.

Reply:
Naming conventions usually build over precedent. We have an existing type who's naming decision is analogous to this I think.

Our .NET provider type is `AzureAppConfigurationProvider`. Configuration providers pull configuration from a source (file, environment variables, key vault) and provide it to the application. This is a configuration provider that pulls configuration from Azure App Configuration.

I still think we should stick with `ApplicationInsightsTelemetryPublisher`. Our concept of telemetry publishers take feature flag evaluation events and publish them to some data collection destination. This publisher is one that publishes to Application Insights.